lib/service/storage/elasticsearch.ts

```diff
@@ -913,11 +1010,19 @@ class ElasticSearch extends Service {
    * @returns {Promise.<{ _id, _version, _source }>}
    */
   async replace(
-    index,
-    collection,
-    id,
-    content,
-    { refresh, userId = null, injectKuzzleMeta = true } = {},
+    index: string,
+    collection: string,
+    id: string,
+    content: JSONObject,
+    {
+      refresh,
+      userId = null,
+      injectKuzzleMeta = true,
+    }: {
+      refresh?: boolean | "wait_for";
+      userId?: string;
+      injectKuzzleMeta?: boolean;
+    } = {},
```

Comment:
```suggestion
    {
      refresh,
      userId = null,
      injectKuzzleMeta = true,
    }: Exclude<KRequestParams, "retryOnConflict" | "timeout"> = {},
```

With new `KRequestParams` type we can compose type like it, to DRY